### PR TITLE
Collect drain statistics and report to wmstats

### PR DESCRIPTION
**Pull request by goughes**
This will incorporate some statistics on the draining process and report them to wmstats with the other agent info.

### CODE REVIEW

File: src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py
Comment (vkuznet) [inline src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py]: this can be combined with logging.info from line 184.

File: src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py
Comment (vkuznet) [inline src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py]: what's the purpose of having class variable instead of instance one?

File: src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py
Comment (vkuznet) [inline src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py]: why this method needs to be a classmethod since it only returns class variable without doing anything. Can you put drainStats into __int__ method and have it as a member of the class?

**Review by vkuznet — CHANGES_REQUESTED** [review]
I also want to add to my comments that code use mixed syntax, in many places it collapse spaces between key and a values, e.g. replace key = value with key=value, but in others it does the opposite. I would recommend to run this PR through pylint and fix code syntax.

**Comment by ericvaandering** [comment]
Has this code changed since Valentin's comments as I didn't notice the `key=value` issue he mentioned. In any case, `key=value` is correct in function calls, `variable = value` is correct in executable code. The pylint and pep8 reports will tell you that.

File: src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py
Comment (goughes) [inline src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py]: AgentStatusWatcher spawns three threads based of the AgentStatusPoller, DrainStatusPoller and ResourceControlUpdater classes.  Each instance of those classes doesn't know about the other.  AgentStatusPoller needs to get variable contents from DrainStatusPoller, hence the class variable.  If there is a better way, I'm open to suggestions.

**Comment by goughes** [comment]
I haven't changed anything.  I will do another review based on your comments.

**Comment by goughes** [comment]
This references #7615 which has problems with the unittests.  Will open new PR based on #7955 and #7775.

**Comment by ticoann** [comment]
are you replacing #7955 and #7775? If you do that could you close the replaced pull requests?

**Comment by goughes** [comment]
I'll wait until #7955 is merged before making the new PR.  I will close #7775 after.

**Pull request closed without merging**
